### ISSUE TITLE
polygonContains should be uniform when testing the verteces of the polygon

### ISSUE THREAD

**Issue by Kcnarf** [d3/d3-polygon#30]
```
var d3Polygon = require("d3-polygon")
var unitSquare = [[0,0], [0,1], [1,1], [1,0]];
var polygonContains = d3Polygon.polygonContains;
unitSquare.forEach( p => console.log(`[${p}] in unitSquare : ${polygonContains(unitSquare, p)}`));

>"[0,0] in unitSquare : true"
>"[0,1] in unitSquare : false"
>"[1,1] in unitSquare : false"
>"[1,0] in unitSquare : false"
```

In Runkit => https://runkit.com/embed/tht1jbzi6iqq

Imho, all the verteces of the polygon should be either inside or outside.

**Comment by mbostock** [comment]
polygonContains [is not robust](https://github.com/mikolalysenko/robust-point-in-polygon), so it’s not guaranteed to return the correct result in edge cases like this. We could tweak the algorithm to make it perform better in special cases, but only at the expense of complexity and speed. I don’t think we want to go all the way to making it robust.

See https://observablehq.com/@tmcw/understanding-point-in-polygon for an explanation of the algorithm.

**Comment by Kcnarf** [comment]
Thanks for the response and the provided links.

I discovered the use case while debugging some tests I did for d3-weighted-voronoi plugin, which sometimes were failing. I was positioning a site at the [1,1] vertex, considered outside of the unit square, and hence re-position the site in a random fashion (hence the rare failure).

Closing the issue as the non-robustness seems assumed.

PS: perhaps a hint in the documentation may help.
